index.js

```diff
@@ -1,47 +1,52 @@
 'use strict';
 const path = require('path');
-const loadJsonFile = require('load-json-file');
-const pathType = require('path-type');
+const fs = require('graceful-fs');
+const normalizePackageData = require('normalize-package-data');
 
-module.exports = (fp, opts) => {
-	if (typeof fp !== 'string') {
-		opts = fp;
-		fp = '.';
-	}
-
-	opts = opts || {};
+const PACKAGE_FILE = 'package.json';
 
-	return pathType.dir(fp)
-		.then(isDir => {
-			if (isDir) {
-				fp = path.join(fp, 'package.json');
+function readJsonFileAsync(file) {
+	return new Promise((resolve, reject) => {
+		fs.readFile(file, 'utf8', (err, data) => {
```

Comment:
Use https://github.com/ljharb/util.promisify and you don't need the `new Promise`